src/tests/system/tests/test_ldap.py

```diff
@@ -52,3 +52,94 @@ def test_ldap__change_password(client: Client, ldap: LDAP, modify_mode: str):
 
     assert client.auth.ssh.password(user, new_pass), "Authentication with new correct password failed"
     assert not client.auth.ssh.password(user, old_pass), "Authentication with old incorrect password did not fail"
+
+
+@pytest.mark.ticket(bz=[795044, 1695574])
+@pytest.mark.parametrize("modify_mode", ["exop", "ldap_modify"])
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__change_password_new_pass_not_match(client: Client, ldap: LDAP, modify_mode: str):
+    """
+    :title: Change password with "ldap_pwmodify_mode" set to @modify_mode, but retyped password not matched
+    :setup:
+        1. Add user to SSSD, set his password
+        2. Allow user to change his password
+        3. Set "ldap_pwmodify_mode"
+        4. Start SSSD
+    :steps:
+        1. Change password to new password, but retyped password is wrong
+    :expectedresults:
+        1. Passwords mismatched, password change is not successful
+    :customerscenario: True
+    """
+    ldap.user("user1").add(password="Secret123")
+    ldap.aci.add('(targetattr="userpassword")(version 3.0; acl "pwp test"; allow (all) userdn="ldap:///self";)')
+
+    client.sssd.domain["ldap_pwmodify_mode"] = modify_mode
+    client.sssd.start()
+
+    assert not client.auth.passwd.password(
+        "user1", "Secret123", "Red123", "Hat000"
+    ), "Password changed successfully, which is not expected"
+
+
+@pytest.mark.ticket(bz=[795044, 1695574, 1795220])
+@pytest.mark.parametrize("modify_mode", ["exop", "ldap_modify"])
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__change_password_all_small(client: Client, ldap: LDAP, modify_mode: str):
+    """
+    :title: Change password to all small letters, password check fail
```

Comment:
@justin-stephenson is `small letters` understandable? I'd use lower case instead, but I'm not native

Reply:
please change to 'lower-case'